x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -2343,7 +2343,13 @@ private void expectMetadataForApiKey(final Map<String, Object> expectedMetadata,
         assertNotNull(actualRawApiKeyDoc);
         @SuppressWarnings("unchecked")
         final var actualMetadata = (Map<String, Object>) actualRawApiKeyDoc.get("metadata_flattened");
-        assertThat("for api key doc " + actualRawApiKeyDoc, actualMetadata, equalTo(expectedMetadata));
+        assertThat(
+            "for api key doc " + actualRawApiKeyDoc,
+            // Internally, metadata may be stored as `null`. However, it is always exposed as an empty map through the API. We define
+            // `expectedMetadata` as the expected value according to the API, so we need to account for this discrepancy here
+            actualMetadata == null ? anEmptyMap() : actualMetadata,
```

Comment:
I'm addressing the issue here, instead of going the other direction and accounting for `null` values on the GET API response. I think it is more intuitive to treat the API response as the expected result. We also run into an issue with null pointers if we try to pass a null value into `expectAttributesForApiKey` via `Map.of(METADATA, null)`.

I would be up for only asserting on the API response, and removing `expectMetadataForApiKey` altogether.